app/models/participant_profile.rb

```diff
@@ -89,4 +89,15 @@ def update_schedule!(schedule)
     # TODO: Do we need to store when this happens outside of papertrail?
     update!(schedule: schedule)
   end
+
+  def validate_participant_state(declaration_date, milestone)
+    if (%w[withdrawn deferred] & participant_profile_states.map(&:state)).any?
+      raise ActionController::ParameterMissing, I18n.t(:declaration_on_incorrect_state) if declaration_date >= Time.zone.now
+
+      active_state_date = participant_profile_states.where(state: "active").order(:created_at).last.created_at
+      unless active_state_date && active_state_date <= milestone.milestone_date.end_of_day
+        raise ActionController::ParameterMissing, I18n.t(:declaration_on_incorrect_state)
+      end
+    end
```

Comment:
I think what this should do is: fetch the last participant state before the declaration date (using `created_at`).

Then it says 'Ok, if that state is active, I let you make the declaration. If it is not, you cannot make the declaration'.

I do not think this should validate that declaration date is not in future, or anything relating to milestones - because this should have been validated already.

Reply:
I think it's good idea. I've pushed the change